docs/modules/ROOT/pages/vshn-managed/postgresql/restore.adoc

```diff
@@ -0,0 +1,78 @@
+= Restore a PostgreSQL service
+
+IMPORTANT: The restores can only be triggered from an instance in the same namespace.
+If the old instance has already been deleted the backups are deleted as well.
+
+. Get the name of the backup you want to restore by listing all available backups.
++
+[source,bash]
+----
+kubectl -n postgres get vshnpostgresbackups
+BACKUP NAME                                 DATABASE INSTANCE     STATUS     AGE
+pgsql-app1-prod-pk8k4-2023-03-05-13-05-00   pgsql-app1-prod       Completed  38s
+pgsql-app1-dev-pk8k4-2023-03-01-16-52-11    pgsql-app1-dev        Failed     28s
+----
+NOTE: The list will show all backups from all instances in this namespace. In the example above there are two database instances with one backup each.
+
+. Choose a specific backup and get its details.
++
+[source,bash]
+----
+kubectl -n postgres get vshnpostgresbackups pgsql-app1-prod-pk8k4-2023-03-05-13-05-00 -oyaml
+----
++
+[source,yaml]
+----
+apiVersion: api.appcat.vshn.io/v1
+kind: VSHNPostgresBackup
+metadata:
+  name: pgsql-app1-prod-pk8k4-2023-03-05-13-05-00
+  namespace: postgres
+status:
+  databaseInstance: pgsql-app1-prod
+  process:
+    failure: "" <1>
+    jobPod: pgsql-app1-prod-pk8k4-backup-27967024-6xjpq
+    status: Completed
+    timing:
+      end: "2023-03-05T13:05:15.61228Z"
+      start: "2023-03-05T13:05:09.519369Z"
+      stored: "2023-03-05T13:05:15.755Z"
+----
+<1> The failure message in case the job fails.
+
+. Apply the following object on your namespace, as specified by its YAML description.
++
+.Example to restore a PostgreSQL instance
+[source,yaml]
+----
+apiVersion: vshn.appcat.vshn.io/v1
+kind: VSHNPostgreSQL
+metadata:
+  name: pgsql-app1-restore
+  namespace: postgres
+spec:
+  parameters:
+    restore:
+      claimName: pgsql-app1-prod <1>
+      backupName: pgsql-app1-prod-pk8k4-2023-03-01-16-52-02 <2>
+    backup:
+      schedule: '0 22 * * *'
+    service:
+      majorVersion: "15"
+      pgSettings:
+        timezone: Europe/Zurich
+    size:
+      cpu: "600m"
+      memory: "3500Mi"
+      disk: "80Gi"
+  writeConnectionSecretToRef:
+    name: postgres-creds-restored
+  compositionRef:
+    name: vshnpostgresrestore.vshn.appcat.vshn.io <3>
+----
+<1> The name of the instance you want to restore from
+<2> The backup name you want to restore
+<3> You need to specify that this is not a normal instance but a restored one
+
+. After the restore the new instance will behave exactly the same as any other PostgreSQL by VSHN instance.
```

Comment:
That's not an actionable step, isn't it? So no need to be in the list. How can I make this object "normal" looking, without the restore stuff in it? Aka how can I make it GitOps-ready again?

Reply:
> That's not an actionable step, isn't it? So no need to be in the list.
I will remove the action point.

> How can I make this object "normal" looking, without the restore stuff in it? Aka how can I make it GitOps-ready again?

If I understood you correctly, for now we cannot make it GitOps ready. This is partly due to the limitations of Crossplane. We are using a second composition for the restore process. @Kidswiss do you see it differently?

Reply:
It might be possible to switch the composition back to the default one after the restore has finished. But haven't tested that.

Reply:
Please test it, sounds like a very important thing. This isn't the "original" object anymore, which would completely break CI/CD pipelines and GitOps.

Reply:
I'm not quite sure what you mean by "original" object.

The restore needs to be done with a new instance defined by a new object anyway. And that new object can also be managed via GitOps.